firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionsFallbackActivityLifecycleCallbacks.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.app.Activity
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Bundle
+import android.util.Log
+import com.google.firebase.annotations.concurrent.Background
+import com.google.firebase.sessions.api.FirebaseSessionsDependencies
+import com.google.firebase.sessions.api.SessionSubscriber
+import com.google.firebase.sessions.settings.SessionsSettings
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@Singleton
+internal class SessionsFallbackActivityLifecycleCallbacks
+@Inject
+constructor(
+  private val sessionsSettings: SessionsSettings,
+  private val sessionGenerator: SessionGenerator,
+  private val timeProvider: TimeProvider,
+  @Background private val backgroundDispatcher: CoroutineContext,
+) {
+  private var localSessionData: SessionData
+  init {
+    localSessionData =
+      SessionData(
+        sessionDetails = sessionGenerator.generateNewSession(null),
+        backgroundTime = timeProvider.currentTime()
+      )
+    if (!sessionsSettings.sessionsEnabled) {
```

Comment:
Still need this setting check, otherwise it will trigger notify subscriber for singleton init